Translation to PT-BR Completed: "removing-effect-dependencies" (#777)
reactjs/pt-br.react.dev: src/content/learn/removing-effect-dependencies.md

    const connection = createConnection(serverUrl, roomId);
    connection.connect();
    return () => connection.disconnect();
  }, []); // <-- Corrija o erro aqui!
  return <h1>Bem-vindo(a) à sala {roomId}!</h1>;
}

export default function App() {

>   const [roomId, setRoomId] = useState('geral');

Reviewer: We avoid translating the code values, it seems like I'm translating a variable hehe
```suggestion
  const [roomId, setRoomId] = useState('general');
```

  return (
    <>
      <label>
         Escolha a sala de bate-papo:{' '}
        <select
          value={roomId}
          onChange={e => setRoomId(e.target.value)}
        >

>           <option value="geral">geral</option>
>           <option value="viagem">viagem</option>
>           <option value="música">música</option>

Reviewer: ```suggestion
          <option value="general">geral</option>
          <option value="travel">viagem</option>
          <option value="music">música</option>
```

    connection.connect();
    return () => connection.disconnect();
  }, [roomId]);
  return <h1>Bem-vindo(a) à sala {roomId}!</h1>;
}

export default function App() {

>   const [roomId, setRoomId] = useState('geral');

Reviewer: ```suggestion
  const [roomId, setRoomId] = useState('general');
```

  return (
    <>
      <label>
        Escolha a sala de bate-papo:{' '}
        <select
          value={roomId}
          onChange={e => setRoomId(e.target.value)}
        >

>           <option value="geral">geral</option>
>           <option value="viagem">viagem</option>
>           <option value="música">música</option>

Reviewer: ```suggestion
          <option value="general">geral</option>
          <option value="travel">viagem</option>
          <option value="music">música</option>
```

import { createConnection } from './chat.js';

const serverUrl = 'https://localhost:1234';

> const roomId = 'música';

Reviewer: ```suggestion
const roomId = 'music';
```

  return (
    <>
      <h1>Bem-vindo(a) à sala {roomId}!</h1>
      <input value={message} onChange={e => setMessage(e.target.value)} />
    </>
  );
}

export default function App() {

>   const [roomId, setRoomId] = useState('geral');

Reviewer: ```suggestion
  const [roomId, setRoomId] = useState('general');
```

  return (
    <>
      <label>
        Escolha a sala de bate-papo:{' '}
        <select
          value={roomId}
          onChange={e => setRoomId(e.target.value)}
        >

>           <option value="geral">geral</option>
>           <option value="viagem">viagem</option>
>           <option value="música">música</option>

Reviewer: ```suggestion
          <option value="general">geral</option>
          <option value="travel">viagem</option>
          <option value="music">música</option>
```

  const [count, setCount] = useState(0);

  useEffect(() => {
    console.log('✅ Criando um intervalo');
    const id = setInterval(() => {
      console.log('⏰ Intervalo de tique-taque');
      setCount(c => c + 1);
    }, 1000);
    return () => {
      console.log('❌ Limpar o intervalo');
      clearInterval(id);
    };
  }, []);

  return <h1>Contador: {count}</h1>
}
```

</Sandpack>

Em vez de ler o valor de `count` dentro do Efeito, você passa uma instrução `c => c + 1` ("incremente esse número!") para o React. O React aplicará essa instrução na próxima renderização. Como você não precisa mais ler o valor de `count` dentro do seu Efeito, você pode manter as dependências do Efeito vazias (`[]`). Isso evita que o Efeito recrie o intervalo a cada tique-taque.

</Solution>

#### Corrigir uma animação que está sendo reiniciada repetidamente {/*fix-a-retriggering-animation*/}

> Neste exemplo, ao pressionar "Mostrar", uma mensagem de boas-vindas aparece gradualmente. A animação leva um segundo. Quando você pressiona "Remover", a mensagem de boas-vindas desaparece imediatamente. A lógica para a animação de fade-in é implementada no arquivo `animation.js` como um [loop de animação](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) em JavaScript. Você não precisa alterar essa lógica, pois pode tratá-la como uma biblioteca de terceiros. Seu Efeito cria uma instância de `FadeInAnimation` para o nó do DOM e depois chama `start(duration)` ou `stop()` para controlar a animação. A `duration` é controlada por um *slider*. Ajuste o *slider* e veja como a animação muda.

Reviewer: ```suggestion
Neste exemplo, ao pressionar "Mostrar", uma mensagem de boas-vindas aparece gradualmente. A animação leva um segundo. Quando você pressiona "Remover", a mensagem de boas-vindas desaparece imediatamente. A lógica para a animação de fade-in é implementada no arquivo `animation.js` como um [loop de animação](https://developer.mozilla.org/pt-BR/docs/Web/API/window/requestAnimationFrame) em JavaScript. Você não precisa alterar essa lógica, pois pode tratá-la como uma biblioteca de terceiros. Seu Efeito cria uma instância de `FadeInAnimation` para o nó do DOM e depois chama `start(duration)` ou `stop()` para controlar a animação. A `duration` é controlada por um *slider*. Ajuste o *slider* e veja como a animação muda.
```
